src/jimgw/core/single_event/transform_utils.py

```
@@ -0,0 +1,620 @@
+    phi_jl = safe_arctan2(LNh[1], LNh[0])
+    phi_jl = (phi_jl + 2 * jnp.pi) % (2 * jnp.pi)  # Ensure 0 <= phi_jl < 2pi
+
+    return theta_jn, phi_jl, tilt_1, tilt_2, phi_12, chi_1, chi_2
+
+
+# =============================================================================
+# Sky / detector coordinate transforms
+# =============================================================================
+
+
+def _rotate_y(angle: Float) -> Float[Array, "3 3"]:
```

Ruff (0.15.12)

[error] Syntax error in forward annotation: Unexpected token at the end of an expression

(F722)